### PR TITLE
feat: custom error message and code for rate-limiting

### PR DESCRIPTION
**Pull request by omegabytes**
Adds user-configurable error_code and error_message fields to the rate-limiting plugin for v3.1^.

### CODE REVIEW

**Comment by aboudreault** [comment]
ref: https://github.com/Kong/kong/pull/8930

File: internal/test/e2e/plugins.go
Comment (GGabriele) [inline internal/test/e2e/plugins.go:282]: While you are at it, would you mind documenting the expected changes for older DPs too? Like that `redis_*` fields are removed for DP < 2.7 etc

**Comment by aboudreault** [comment]
LGTM Alex, let's wait for another approval to make sure I am not missing something.

File: internal/test/e2e/plugins.go
Comment (omegabytes) [reply internal/test/e2e/plugins.go:286]: [cfa8dd2](https://github.com/Kong/koko/pull/515/commits/cfa8dd288786e9cb4951be2d4f6a3e007e9e5a62)

Also noticed that `redis_password` is not being removed? https://github.com/Kong/koko/blob/69228016c6c464471b8ea53bb5426677e073f2ff/internal/server/kong/ws/config/compat/compatibility_table.go#L382-L398

Comment (aboudreault) [reply internal/test/e2e/plugins.go:286]: that seems correct. `redis_password` was added in a previous kong version. ref: https://github.com/Kong/kong/pull/8032

**Review by GGabriele — APPROVED** [review]

**Pull request merged**
